pkg/ruler/ruler.go

```diff
@@ -189,6 +192,9 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet, logger log.Logger) {
 
 	f.BoolVar(&cfg.EnableQueryStats, "ruler.query-stats-enabled", false, "Report the wall time for ruler queries to complete as a per-tenant metric and as an info level log message.")
 
+	f.Int64Var(&cfg.MaxIndependentRuleEvaluationConcurrency, "ruler.max-independent-rule-evaluation-concurrency", 0, "Number of rules rules that don't have dependencies that we allow to be evaluated concurrently across all tenants. 0 to disable.")
+	f.Float64Var(&cfg.ThresholdIndependentRuleEvaluationConcurrency, "ruler.threshold-independent-rule-evaluation-concurrency", 50.0, "Threshold of the interval to last rule group runtime duration to allow a rule to be evaluated concurrency. Expressed in terms of percentage and as a float. By default, the rule group runtime duration must exceed 50.0% of the evaluation interval.")
```

Comment:
[nit, non blocking] In other configs we add the `percentage` suffix to clarify the unit. Have you considered something like `ruler.independent-rule-evaluation-concurrency-min-duration-percentage`?

Reply:
Done in https://github.com/grafana/mimir/pull/8146/commits/c4f16e748b7e8d28b45645f82a025ba999ec5a4a